faiss/gpu/test/CMakeLists.txt

```diff
@@ -17,12 +17,15 @@
 # the License.
 # =============================================================================
 
-find_package(CUDAToolkit REQUIRED)
-
 # Defines `gtest_discover_tests()`.
 include(GoogleTest)
 add_library(faiss_gpu_test_helper TestUtils.cpp)
-target_link_libraries(faiss_gpu_test_helper PUBLIC faiss gtest CUDA::cudart $<$<BOOL:${FAISS_ENABLE_RAFT}>:raft::raft> $<$<BOOL:${FAISS_ENABLE_RAFT}>:raft::compiled>)
+if(USE_ROCM)
+  target_link_libraries(faiss_gpu_test_helper PUBLIC faiss gtest $<$<BOOL:${USE_ROCM}>:hip::host>)
+else()
+  find_package(CUDAToolkit REQUIRED)
+  target_link_libraries(faiss_gpu_test_helper PUBLIC faiss gtest CUDA::cudart $<$<BOOL:${FAISS_ENABLE_RAFT}>:raft::raft> $<$<BOOL:${FAISS_ENABLE_RAFT}>:raft::compiled>)
+endif()
```

Comment:
@iotamudelta @ItsPitt This PR breaks the TestCodePacking tests internally. While I am looking at it, do you think we remove collapse these if-else statements throughout the CMakeLists files where possible? That was the intent behind my original comment to use the inline condition includes.

Reply:
@ramilbakhshyiev Just to clarify, are these if-else conditions breaking the TestCodePacking, or is that a separate issue?

Reply:
Separate issue. I just didn't notice these yesterday when I was going over the diff after you addressed the comments, my apologies.

Reply:
If you're curious, here are the errors I am getting:
```
Faiss assertion 'err__ == cudaSuccess' failed in int faiss::gpu::getCurrentDevice() at fbcode/faiss/gpu/utils/DeviceUtils.cu:23; details: CUDA error 35 CUDA driver version is insufficient for CUDA runtime version
```

I think that's because of the call to get warp size from the device properties. I'm trying to understand why that fails internally.

Reply:
I figured out the internal error and published a change for that. It should fix this PR failures internally.